docs: update pull_request.md and PULL_REQUEST_TEMPLATE.md (#6106)
codeigniter4/CodeIgniter4: contributing/pull_request.md

All code must conform to our [Style Guide](./styleguide.md), which is
based on PSR-12.

This makes certain that all submitted code is of the same format
as the existing code and ensures that the codebase will be as readable as possible.

You can fix most of the coding style violations by running this command in your terminal:

```
> composer cs-fix
```

You can check the coding style violations:

> ```

Reviewer: Specify `bash` formatting?

Author: Added `console`.

https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks#syntax-highlighting
https://github.com/github/linguist/blob/d7799da826e01acdb8f84694d33116dccaabe9c2/lib/linguist/languages.yml#L6063

Reviewer: Wasn't aware of "ShellSession", very cool.

### Branching

All bug fixes should be sent to the __"develop"__ branch, this is where the next bug fix version will be developed.

> PRs with any enhancement should be sent to __"4.3"__ branch, this is where the next minor upgrade version will be developed.

Reviewer: ```suggestion
PRs with any enhancement should be sent to next minor version branch, e.g. __"4.3"__
```
